src/main/kotlin/com/deepromeet/atcha/transit/infrastructure/client/public/common/CompositeBusPositionFetcher.kt

```
@@ -1,26 +1,26 @@
-    private val busPositionFetcherMap: Map<ServiceRegion, BusPositionFetcher>
+    private val busPositionFetchers: BusPositionFetchers
 ) {
     suspend fun fetch(busRoute: BusRoute): List<BusPosition> {
         if (busRoute.serviceRegion == ServiceRegion.SEOUL) {
             return try {
-                val result = busPositionFetcherMap[busRoute.serviceRegion]!!.fetch(busRoute.id)
-                if (result.isEmpty()) return busPositionFetcherMap[ServiceRegion.GYEONGGI]!!.fetch(busRoute.id)
+                val result = busPositionFetchers.forRegion(busRoute.serviceRegion).fetch(busRoute.id)
+                if (result.isEmpty()) return busPositionFetchers.forRegion(ServiceRegion.GYEONGGI).fetch(busRoute.id)
                 result
             } catch (e: Exception) {
```

detekt (1.23.8)

[warning] The caught exception is swallowed. The original exception could be lost.

(detekt.exceptions.SwallowedException)